### PR TITLE
Quick Register for event

### PR DESCRIPTION
**Pull request by dchen150**
- user is PATCHed if they exist in the backend
- these components will be used from the EventDetails page

NOTE: backend for PATCH is erroring (400) due to https://stackoverflow.com/questions/48653365/update-attribute-timestamp-reserved-word, going to look into updating backend ASAP (edit: complete and merged in the backend)

image

image

### CODE REVIEW

**Comment by dchen150** [comment]
need to change 'them/them/their' to 'they/them/their'

File: src/pages/member/QuickRegister.js [hunk 1]
```diff
@@ -82,36 +109,43 @@ const EventFormContainer = (props) => {
     return (
       // assumes that the event details component that uses this component (QuickRegister) does not allow
       // the user to get to this page if they are already signed up or the event has passed
```
Comment (superzzp) [inline src/pages/member/QuickRegister.js:111]: So we will allow users to unregister an event in the eventDetails page right? And the users will be allowed to get to this page again after unregistering the event

Comment (dchen150) [reply src/pages/member/QuickRegister.js:111]: yes, i think so.

File: src/pages/member/QuickRegister.js [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+        {isSignedUp
+          ? <div className={classes.layout}>
+            <Paper className={classes.paper}>
+              <div className={classes.completeContainer}>
+                <div className={classes.message}>
+                  <div className={classes.houseContainer}>
+                    <img src={House} className={classes.house} alt='BizTech House' />
+                  </div>
+                  <Typography className={classes.done}>done!</Typography>
+                  <Typography>you are now registered, click <strong onClick={handleReturn} style={{ cursor: 'pointer' }}>here</strong> <br/> to return to the previous menu.</Typography>
```
Comment (ianmah) [inline src/pages/member/QuickRegister.js:125]: maybe "previous page" is better

File: src/pages/member/QuickRegister.js [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+  async function registerUser (id, eventID, heardFrom) {
+    const body = {
+      id,
+      eventID,
+      heardFrom,
+      registrationStatus: 'registered'
+    }
+    fetchBackend('/registrations', 'POST', body)
+      .then((regResponse) => {
+        alert('Signed Up')
```
Comment (ianmah) [inline src/pages/member/QuickRegister.js:227]: Do we need this if they see the Done screen afterwards?

Comment (dchen150) [reply src/pages/member/QuickRegister.js:227]: good point

**Pull request merged**